Doppler.Sap.Test/Utils/DateTimeProviderTest.cs

```diff
@@ -0,0 +1,102 @@
+            };
+            var dateTimeProvider = new DateTimeProvider();
+            var utcDate = dateTimeProvider.UtcNow;
+            var cstZone = TimeZoneInfo.FindSystemTimeZoneById(timeZoneConfigurations.InvoicesTimeZone);
+
+            var expectedDate = utcDate.AddMinutes(cstZone.BaseUtcOffset.TotalMinutes);
+            var result = dateTimeProvider.GetDateByTimezoneId(utcDate, timeZoneConfigurations.InvoicesTimeZone);
+
+            Assert.Equal(expectedDate, result);
+        }
+
+        [Fact]
+        public void DateTimeProvider_ShouldBeReturnPreviousDateInArgentinaTimezone_WhenTimeZoneIsNotNullAndTheDateAfter0clock()
+        {
+            var timeZoneConfigurations = new TimeZoneConfigurations
+            {
+                InvoicesTimeZone = TimeZoneHelper.GetTimeZoneByOperativeSystem("Argentina Standard Time")
+            };
+            var dateTimeProvider = new DateTimeProvider();
+            var utcDate = new DateTime(2020, 12, 13, 0, 10, 0);
+            var expectedDate = utcDate.AddDays(-1);
+
+            var result = dateTimeProvider.GetDateByTimezoneId(utcDate, timeZoneConfigurations.InvoicesTimeZone);
+
+            Assert.Equal(expectedDate.Date, result.Date);
+        }
+
+        [Fact]
+        public void DateTimeProvider_ShouldBeReturnDate_WhenTimeZoneIsNull()
+        {
+            var timezoneId = "";
+            var dateTimeProvider = new DateTimeProvider();
+            var utcDate = dateTimeProvider.UtcNow;
+
+            var result = dateTimeProvider.GetDateByTimezoneId(utcDate, timezoneId);
+
+            Assert.Equal(utcDate, result);
+        }
+
+        [Fact]
+        public void DateTimeProvider_PredefinedValues_ShouldBeReturnDateInArgentinaTimezone_WhenTimeZoneIsNotNull()
+        {
+            var timeZoneConfigurations = new TimeZoneConfigurations
+            {
+                InvoicesTimeZone = TimeZoneHelper.GetTimeZoneByOperativeSystem("Argentina Standard Time")
+            };
+            var dateTimeProvider = new DateTimeProvider();
+            var utcDate = new DateTime(2020, 12, 14, 14, 0, 0, DateTimeKind.Utc);
+            var expectedDate = new DateTime(2020, 12, 14, 11, 0, 0);
+
+            var result = dateTimeProvider.GetDateByTimezoneId(utcDate, timeZoneConfigurations.InvoicesTimeZone);
+
+            Assert.Equal(expectedDate, result);
+        }
+
+        [Fact]
+        public void DateTimeProvider_PredefinedValues_ShouldBeReturnPreviousDateInArgentinaTimezone_WhenTimeZoneIsNotNullAndTheDateAfter0clockAndKingUtc()
+        {
+            var timeZoneConfigurations = new TimeZoneConfigurations
+            {
+                InvoicesTimeZone = TimeZoneHelper.GetTimeZoneByOperativeSystem("Argentina Standard Time")
+            };
+            var dateTimeProvider = new DateTimeProvider();
+            var date = new DateTime(2020, 12, 13, 0, 10, 0, DateTimeKind.Utc);
```

Comment:
You are testing with `DateTimeKind.Utc` and `DateTimeKind.Unspecified`, is `DateTimeKind.Local` missing?

Reply:
For the DateTimeKind.Local I had an exception:
``` System.ArgumentException : The conversion could not be completed because the supplied DateTime did not have the Kind property set correctly.  For example, when the Kind property is DateTimeKind.Local, the source time zone must be TimeZoneInfo.Local. (Parameter 'sourceTimeZone')```

I am investigating it.

Reply:
I think that it is fine, it can be kept as it is because that is not an expected scenario.

What do you think about making it clear and add a test expecting this exception?

Reply:
Yes, sure, currently I am creating a unit test to validate an ```ArgumentException``` when the kind is Local.

Reply:
Added a new test to validate the exception when the ```Kind is Local```